pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -45,10 +52,11 @@
 
 
 /**
+ * An implementation of {@link PinotConfigProvider}
  * The {@code TableCache} caches all the table configs and schemas within the cluster, and listens on ZK changes to keep
  * them in sync. It also maintains the table name map and the column name map for case-insensitive queries.
  */
-public class TableCache {
+public class TableCache implements PinotConfigProvider {
```

Comment:
Maybe good to rename this as DefaultConfigProvider (or some such), but perhaps that is better done in another PR.